crates/subspace-networking/src/node_runner.rs

```diff
@@ -690,22 +687,26 @@ where
         let local_peer_id = *self.swarm.local_peer_id();
 
         if let IdentifyEvent::Received { peer_id, mut info } = event {
-            debug!(?peer_id, protocols=?info.protocols, "IdentifyEvent::Received");
+            debug!(?peer_id, protocols = ?info.protocols, "IdentifyEvent::Received");
 
             // Check for network partition
             if info.protocol_version != self.protocol_version {
                 debug!(
                     %local_peer_id,
                     %peer_id,
-                    local_protocol_version=%self.protocol_version,
-                    peer_protocol_version=%info.protocol_version,
-                    "Peer has different protocol version. Peer was banned.",
+                    local_protocol_version = %self.protocol_version,
+                    peer_protocol_version = %info.protocol_version,
+                    "Peer has different protocol version, banning temporarily",
                 );
 
-                self.ban_peer(peer_id);
+                self.temporary_bans.lock().create_or_extend(&peer_id);
```

Comment:
I don't think that's enough. Consider reproducing parts of the `ban_peer` logic.

Reply:
What about now?

Reply:
Better, I just remembered that `ban_peer` from libp2p disconnects immediately. I would go to the method implementation and copy the related logic as well.

Reply:
Added disconnection, but now I'm wondering if identify will manage to send a message to the other side in all cases.